scripts/migration/migrate_backrefs_between_nodes_and_nodelogs.py

```diff
@@ -0,0 +1,73 @@
+"""
+This migration will add original_node and node associated with the log to nodelogs. It will then make
+copies of each nodelog for the remaining nodes in the backref (registrations and forks),
+changing the node to the current node.
+"""
+
+import sys
+import logging
+from modularodm import Q
+from website.app import init_app
+from website import models
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+logger = logging.getLogger(__name__)
+
+
+def lookup_node(node_id):
+    """
+    Retrieves original node on nodelog
+    """
+    return models.Node.find(Q('_id', 'eq', node_id))[0]
+
+
+def main(dry=True):
+    init_app(routes=False)
+    node_logs = models.NodeLog.find(Q('original_node', 'eq', None))
+    total_log_count = node_logs.count()
+    count = 0
+    errored_logs = []
+    logger.info('Adding original_node and node fields to each log and then cloning the log for remaining nodes in the log backref.')
+
+    for log in node_logs:
+        # Two steps. Step 1: Add original_node and node field to log.
+        count += 1
+        original_node = lookup_node(log.params['node'])
+        log.original_node = original_node
+        log.node = lookup_node(log._backrefs['logged']['node']['logs'][0])
+        try:
+            log.save()
+            logger.info('{}/{} Log {} "original_node = {} and node = {}" added'.format(count, total_log_count, log._id, log.original_node._id, log.node._id))
+        except KeyError as error:
+            logger.error('Could not migrate log due to error')
+            logger.exception(error)
+            errored_logs.append(log)
+
+        # Step 2: For remaining nodes in log's backrefs, clone log, and point original_node and node fields in the right direction.
+        errored_clones = []
+        backref_total = len(log._backrefs['logged']['node']['logs']) - 1
+        backref_count = 0
+        for node in log._backrefs['logged']['node']['logs'][1:]:
+            backref_count += 1
+            clone = log.clone_node_log(node)
+            clone.original_node = original_node
+            try:
+                clone.save()
+                logger.info('{}/{} {}/{} ...Cloning log {} for node {}. New log is {}'.format(count, total_log_count, backref_count, backref_total, log._id, node, clone._id))
+            except KeyError as error:
+                logger.error('Could not copy node log due to error')
+                logger.exception(error)
+                errored_clones.append([log, clone])
+
+    if dry:
+        raise RuntimeError('Dry run -- transaction rolled back')
+
+
+if __name__ == '__main__':
+
+    dry_run = 'dry' in sys.argv
```

Comment:
This should probably be `'--dry' in sys.argv`. Flags are usually passed in with a single dash for a single letter, or doubly-dashed for a word. 

On an unrelated note, I know this migration works because I accidentally hard-migrated my db. NodeLogs were still persisting and rendering properly after :+1: